charts/spire/charts/spire-server/templates/_helpers.tpl

```diff
@@ -111,6 +111,14 @@ Create the name of the service account to use
 {{- end }}
 {{- end }}
 
+{{- define "spire-server.cluster-name" }}
+{{- if ne (len .Values.clusterName) 0 }}
+{{- .Values.clusterName }}
+{{- else }}
+{{- .Values.global.spire.clusterName }}
+{{- end }}
+{{- end }}
```

Comment:
Can't we do this with the `dig` approach.

So if global is set we take `global` else clustername.

This allows us to still have the defaults when we later on make spire-server and spire-agent a root level chart. That way I think it might work as well without having to change defaults for root level charts.